### PR TITLE
Create namespace modal

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**
The namespace isn't actually created after *Submit* button is pressed. To be done in #1296.

Changes proposed in this pull request:

- Temporarily duplicate `ModalWithForm` and `CustomPropTypes` (to be removed in #1297 )(**you don't have to review it**)
- Add `CreateNamespaceForm` and use it in `ModalWithForm` (working validation)
- Update Fundamental-styles library (globally)
- Move almost all dependencies of core-ui upwards
- add core-ui to gulpfile
- add package-lock.json to core-ui (was missing)

**Related issue(s)**

### CODE REVIEW

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.wrapper {
```
Comment (Wawrzyn321) [inline core-ui/src/components/LabelSelectorInput/LabelSelectorInput.scss]: Please change this name to something a bit less generic.

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  border-width: 1px;
+  border-color: var(--fd-forms-border-color);
+  background-color: var(--fd-forms-background-color);
+  border-radius: 4px;
+  transition: border-color 125ms;
+  padding-left: 8px;
+  align-items: center;
+  min-height: var(--fd-forms-height);
+  flex-wrap: wrap;
+  display: flex;
```
Comment (Wawrzyn321) [inline core-ui/src/components/LabelSelectorInput/LabelSelectorInput.scss]: Would you kindly group these rules to improve readability?

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          <NonRemovableLabel key={l} text={l} />
+        ))}
+
+        {labels.map(l => (
+          <Label key={l} text={l} onClick={() => handleLabelRemoved(l)} />
+        ))}
+        <input
+          className="fd-form__control"
+          type="text"
+          id="input-1"
```
Comment (Wawrzyn321) [inline core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js]: It isn't used anywhere.

File: core-ui/src/setupTests.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import 'jsdom-worker-fix';
 var nodeCrypto = require('crypto');
 global.crypto = {
   getRandomValues: function(buffer) {
-    return nodeCrypto.randomFillSync(new Buffer(buffer));
+    return nodeCrypto.randomFillSync(new Buffer.from(buffer));
```
Comment (y-kkamil) [inline core-ui/src/setupTests.js]: i have also changed it according to: https://nodejs.org/api/buffer.html#buffer_buffer_from_buffer_alloc_and_buffer_allocunsafe

File: core-ui/src/components/CreateNamespaceForm/CreateNamespaceForm.js [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+} from 'fundamental-react';
+import './CreateNamespaceForm.scss';
+import LabelSelectorInput from '../LabelSelectorInput/LabelSelectorInput';
+
+const LIMIT_REGEX =
+  '^[+]?[0-9]*(.[0-9]*)?(([eE][-+]?[0-9]+(.[0-9]*)?)?|([MGTPE]i?)|Ki|k|m)?$';
+
+const ISTIO_INJECTION_LABEL = 'istio-injection=disabled';
+
+const NameField = ({ reference }) => (
```
Comment (y-kkamil) [inline core-ui/src/components/CreateNamespaceForm/CreateNamespaceForm.js]: maybe we can move this to a separate file? as well as `DisableSidecarField`, `MemoryQuotasCheckbox` etc?

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import './LabelSelectorInput.scss';
+import { Token } from 'fundamental-react/Token';
+
+//TODO: move this component to a shared "place"
+
+const labelRegexp = /[a-z0-9A-Z-_.]+=[a-z0-9A-Z-_.]+/;
```
Comment (y-kkamil) [inline core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js]: i believe it is not right, as `examplelabel=` is a proper label in kubernetes world, and does not match this regular expression. see: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

File: core-ui/src/components/ModalWithForm/test/ModalWithForm.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
```
Comment (y-kkamil) [inline core-ui/src/components/ModalWithForm/test/ModalWithForm.test.js]: why not enzyme?

File: core-ui/src/components/typechecking/test/CustomPropTypes.test.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+    it('Returns an error for string', () => {
+      const props = {
+        testprop: 'somestring',
+      };
+
+      expect(
+        CustomPropTypes.elementRef(props, 'testprop', 'testcomponent'),
+      ).toMatchSnapshot();
```
Comment (y-kkamil) [inline core-ui/src/components/typechecking/test/CustomPropTypes.test.js]: i don't get it, why do we do snapshot testing for just pure logic? there's no component involved here, and a simple test would suffice, i think

File: core-ui/src/components/ModalWithForm/ModalWithForm.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+          setOpenStatus(true);
+        }}
+      >
+        {button.text}
+      </Button>
+      <Modal
+        {...props}
+        show={isOpen}
+        actions={
+          <React.Fragment>
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core-ui/src/components/ModalWithForm/ModalWithForm.js]: It's a leftover from the attempt of moving this component to React components.

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js (hunk 6 above)
Comment (sjanota) [reply core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js]: Should be fixed now.

File: core-ui/src/components/ModalWithForm/test/ModalWithForm.test.js (hunk 7 above)
Comment (sjanota) [reply core-ui/src/components/ModalWithForm/test/ModalWithForm.test.js]: I'd say renderer is more stable for full DOM rendering. enzyme lacks support for some newest react features.

File: core-ui/src/components/CreateNamespaceForm/CreateNamespaceForm.js (hunk 5 above)
Comment (sjanota) [reply core-ui/src/components/CreateNamespaceForm/CreateNamespaceForm.js]: I extracted `NameField` to a separate file, but others are not generic enough to become public.

File: core-ui/src/components/typechecking/test/CustomPropTypes.test.js (hunk 8 above)
Comment (sjanota) [reply core-ui/src/components/typechecking/test/CustomPropTypes.test.js]: I refactored it a little. Have a look now.

File: core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js (hunk 6 above)
Comment (y-kkamil) [reply core-ui/src/components/LabelSelectorInput/LabelSelectorInput.js]:
```suggestion
const labelRegexp = /[a-z0-9A-Z-_.]+=[a-z0-9A-Z-_.]*/;
```
still, it doesn't accept `something=` :( i believe the `+` is a problem here. take a look at my suggestion

**Pull request merged**